src/odin/ingestion/afc/afc_archive.py

```diff
@@ -340,6 +352,7 @@ def sync_parquet(self) -> None:
 
         # Check for sigterm before upload (can't be un-done)
         sigterm_check()
+        delete_objects(del_objs)
```

Comment:
Also, I feel like the `if len(objects) == 0:` should be here. It's a little bit clearer that we are only deleting if there are objects.

Reply:
There are no issues with passing a list of 0 elements into `delete_objects`. Several other parts of the application do that regularly.

Previously, passing [] would just cause a ThreadPool to spin up, do nothing, and then shut down, with some associated logging output. Adding the `len(object)==0` check to `delete_objects` mostly just saves some unnecessary logging output.